src/pages/CartPage/Cart.jsx

```diff
@@ -1,6 +1,142 @@
 import "./Cart.css";
-
+import axios from "axios";
+import {useCart} from "../../hooks/useCart";
+import {useToken} from "../../hooks/useToken";
+import NavBar from "../../components/NavBar/NavBar"
 export const Cart=()=>{
-
+    const {state,dispatch}= useCart();
+    const cart=[...state.cartData];
+    const {token}= useToken();
+    const encodedToken=token;
+  
+    const removeHandler = async (productId) => {
+   
+      const response = await axios.delete(
+        `/api/user/cart/${productId}`,
+        {
+            headers:{
+              authorization: encodedToken,
+          },
+        }
+      );
+     
+    dispatch({type:"ADD_TO_CART", payload:response.data.cart});
+    };
+  
+    const increaseCountHandler = (id) => {
+      dispatch({ type:"PRODUCT_COUNT", payload:
+        cart.map((item) =>
+          item._id === id ? { ...item, quantity: item.quantity + 1 } : item
+        )}
+      );
+    };
+  
+    const decreaseCountHandler = (id) => {
+        dispatch({ type:"PRODUCT_COUNT", payload:
+        cart.map((item) =>
+          item._id === id ? { ...item, quantity: item.quantity - 1 } : item
```

Comment:
very neat and organized code, just one suggestion handle the `decreaseCountHandler` case when count is less than 1.